cxx-checks/src/main/java/org/sonar/cxx/checks/TooManyStatementsPerLineCheck.java

```diff
@@ -109,6 +109,11 @@ private boolean isEmptyExpressionStatement(AstNode astNode) {
       if (statement != null ) {
         return astNode.getTokenLine() == statement.getTokenLine();
       }
+
+      AstNode prevNode = astNode.getPreviousAstNode();
```

Comment:
@tomassdepakin check for prevNode == null missing? Or better using isGeneratedNodeExcluded?